dissolve-struct-lib/src/main/scala/ch/ethz/dalab/dissolve/classification/BinarySVMWithDBCFW.scala

```diff
@@ -118,9 +119,15 @@ object BinarySVMWithDBCFW extends DissolveFunctions[Vector[Double], Double] {
       data.map {
         case x: LabeledPoint =>
           new LabeledObject[Vector[Double], Double](x.label,
-            if (solverOptions.sparse)
-              SparseVector(x.features.toArray)
-            else
+            if (solverOptions.sparse){
+              val features:Vector[Double] =  x.features match{
+                case features:org.apache.spark.mllib.linalg.SparseVector =>
+                  val builder:VectorBuilder[Double] = new VectorBuilder(features.indices,features.values,features.indices.length,x.features.size)
+                  builder.toSparseVector
+                case _ => SparseVector(x.features.toArray)
+              } 
+              features  
+            }else
```

Comment:
Yes, since this avoids calling x.features.toArray which in case of very long feature vecors would create a massive overhead. This part was the cause of some OutOfMemory exceptions. With 10k features this part would take forever. However with this the conversion is much faster.